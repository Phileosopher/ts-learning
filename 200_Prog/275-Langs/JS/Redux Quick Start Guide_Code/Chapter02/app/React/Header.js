import React from "react";
import Link from "react-router-dom/Link";

const Header = () => (
  <ul>
    <li>
      <Link to="/">Home</Link>
    </li>
    <li>
      <Link to="/about">Services</Link>
    </li>
    <li>
      <Link to="/topics">Contact Us</Link>
    </li>
    <li>
      <Link to="/topics">Login</Link>
    </li>
  </ul>
);

export default Header;
