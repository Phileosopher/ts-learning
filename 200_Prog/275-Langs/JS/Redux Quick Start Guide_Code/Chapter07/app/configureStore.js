import { createStore, applyMiddleware, compose } from 'redux';
import { fromJS } from 'immutable';
import { routerMiddleware } from 'connected-react-router/immutable';
import createSagaMiddleware from 'redux-saga';
import createReducer from './reducers';

const sagaMiddleware = createSagaMiddleware();

export default function configureStore(initialState = {}, history) {
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  const enhancers = [applyMiddleware(...middlewares)];

  /* eslint-disable no-underscore-dangle, indent */
  const composeEnhancers =
    process.env.NODE_ENV !== 'production' &&
    typeof window === 'object' &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
      : compose;

  const store = createStore(
    createReducer(),
    fromJS(initialState),
    composeEnhancers(...enhancers),
  );

  store.runSaga = sagaMiddleware.run;
  store.injectedReducers = {};
  store.injectedSagas = {};

  return store;
}
