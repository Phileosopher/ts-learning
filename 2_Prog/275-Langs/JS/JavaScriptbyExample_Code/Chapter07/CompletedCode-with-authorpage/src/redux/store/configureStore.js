import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { autoRehydrate } from 'redux-persist';

import rootReducer from '../reducers/rootReducer';

const configureStore = (preloadedState) => {
  return createStore(
    rootReducer,
    preloadedState,
    applyMiddleware(thunk),
    autoRehydrate()
  );
};

export default configureStore;
