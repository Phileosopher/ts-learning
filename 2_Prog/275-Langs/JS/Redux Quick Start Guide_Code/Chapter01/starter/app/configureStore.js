import { createStore } from 'redux';

import createReducer from './reducers';

export default function configureStore(initialState = {}, history) {
  const store = createStore(createReducer());

  // Extensions
  store.injectedReducers = {}; // Reducer registry

  return store;
}
