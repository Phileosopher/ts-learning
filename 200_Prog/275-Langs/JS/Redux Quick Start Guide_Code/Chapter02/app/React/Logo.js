import React from "react";
import Link from "react-router-dom/Link";

const Logo = () => (
  <div>
    <Link to="/">FastDoctor</Link>
  </div>
);

export default Logo;
