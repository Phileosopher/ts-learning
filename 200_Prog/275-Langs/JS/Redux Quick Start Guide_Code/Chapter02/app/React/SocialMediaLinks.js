import React from "react";
import Link from "react-router-dom/Link";

const SocialMediaLinks = () => (
  <ul>
    <li>
      <Link to="https://www.facebook.com/jagapati">Facebook</Link>
    </li>
    <li>
      <Link to="https://github.com/sureshHARDIYA">GitHub</Link>
    </li>
    <li>
      <Link to="https://www.researchgate.net/profile/Suresh_Mukhiya2">
        ResearchGate
      </Link>
    </li>
    <li>
      <Link to="https://twitter.com/dr_code_skm">Twitter</Link>
    </li>
  </ul>
);

export default SocialMediaLinks;
