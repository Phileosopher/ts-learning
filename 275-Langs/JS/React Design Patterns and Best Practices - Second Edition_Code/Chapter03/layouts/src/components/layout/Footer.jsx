import React from 'react';
import { string } from 'prop-types';

const Footer = ({ copyright }) => (
  <div className="Footer">
    <p>&copy; {copyright}</p>
  </div>
);

Footer.propTypes = {
  copyright: string.isRequired
};

export default Footer;
