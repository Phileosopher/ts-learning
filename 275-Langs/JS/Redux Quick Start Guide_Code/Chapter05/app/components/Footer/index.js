import React from 'react';

import A from 'components/A';
import Wrapper from './Wrapper';

function Footer() {
  return (
    <Wrapper>
      <section>This project is licensed under the MIT license.</section>
      <section>
        Made with love by :
        <A href="https://twitter.com/dr_code_skm">S.K. Mukhiya</A>
      </section>
    </Wrapper>
  );
}

export default Footer;
