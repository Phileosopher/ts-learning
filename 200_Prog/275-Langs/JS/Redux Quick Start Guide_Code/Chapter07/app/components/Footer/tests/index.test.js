import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { FormattedMessage, defineMessages } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import Footer from '..';
import ConnectedLanguageProvider from '../../../containers/LanguageProvider';
import configureStore from '../../../configureStore';

import { translationMessages } from '../../../i18n';

const messages = defineMessages({
  licenseMessage: {
    id: 'boilerplate.components.Footer.license.message',
    defaultMessage: 'This project is licensed under the MIT license.',
  },
});

describe('<Footer />', () => {
  let store;
  const Component = (props = {}) =>
    mount(
      <Provider store={store}>
        <ConnectedLanguageProvider messages={translationMessages}>
          <Footer {...props} />
        </ConnectedLanguageProvider>
      </Provider>,
    );

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <ConnectedLanguageProvider messages={translationMessages}>
            <Footer />
          </ConnectedLanguageProvider>
        </Provider>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the default language messages', () => {
    const mountComponent = Component();

    expect(
      mountComponent.contains(
        <FormattedMessage {...messages.licenseMessage} />,
      ),
    ).toBe(true);
  });

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = Component({ attribute: 'test' });
    expect(renderedComponent.find('div').prop('attribute')).toBeUndefined();
  });
});
