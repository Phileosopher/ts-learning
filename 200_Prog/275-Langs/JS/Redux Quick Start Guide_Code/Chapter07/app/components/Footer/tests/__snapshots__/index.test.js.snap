// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the snapshot 1`] = `
<footer
  className="Wrapper-sc-1xi8z6h-0 dCgGdT"
>
  <section>
    <span>
      This project is licensed under the MIT license.
    </span>
  </section>
  <section>
    <div
      className="Wrapper-zn7dhi-0 kmCCBJ"
    >
      <select
        className="Select-wk5r4h-0 hHVceO"
        onChange={[Function]}
        value="en"
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="no"
        >
          no
        </option>
      </select>
    </div>
  </section>
  <section>
    <span>
      Made with love by 
      <a
        href="https://twitter.com/dr_code_skm"
      >
        S.K. Mukhiya
      </a>
      .
    </span>
  </section>
</footer>
`;
